x/payment/genesis_test.go

```diff
@@ -0,0 +1,25 @@
+package payment_test
+
+import (
+	"testing"
+
+	keepertest "github.com/celestiaorg/celestia-app/testutil/keeper"
+	"github.com/celestiaorg/celestia-app/testutil/nullify"
+	"github.com/celestiaorg/celestia-app/x/payment"
+	"github.com/celestiaorg/celestia-app/x/payment/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenesis(t *testing.T) {
+	genesisState := types.GenesisState{
+		Params: types.DefaultParams(),
+	}
+
+	k, ctx := keepertest.PaymentKeeper(t)
+	payment.InitGenesis(ctx, *k, genesisState)
+	got := payment.ExportGenesis(ctx, *k)
+	require.NotNil(t, got)
+
+	nullify.Fill(&genesisState)
+	nullify.Fill(got)
```

Comment:
[question] There aren't any assertions made after these lines and the test passes without them so I'm curious why they are necessary?

Since the nullify package introduces confusing reflection and unsafe code, is it possible to delete entirely?

Reply:
It's a check to make sure that they are copied, iirc

Reply:
Sorry but I don't understand. Can you please elaborate? To make sure what is copied? Happy to hop on a call if you'd prefer that

Reply:
Following the comment in here https://github.com/celestiaorg/celestia-app/pull/893#discussion_r1010285425 (don't want to discuss the test in two separate places). 
I guess what this test is doing is checking whether the exported genesis is correct.
And considering the parameters that we're passing to it:
```go
        genesisState := types.GenesisState{
		Params: types.DefaultParams(),
	}

	k, ctx := keepertest.PaymentKeeper(t)
}
```
And the expected behavior for the genesis export is to have the default genesis (which is already tested in other modules we're importing) + payment params that we just set.
I guess, we will only need the following assertion:
```go
require.Equal(t, defaultParams, got.Params)
```
And, we can get rid of the `Fill` usage